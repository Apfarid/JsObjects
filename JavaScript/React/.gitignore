foo
basic
food-lookup-demo
!components/
