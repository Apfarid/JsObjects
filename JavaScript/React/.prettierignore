**/build/**
**/serviceWorker.js
**/bower_components/**
**/materialize.min.js
**/elf-gatsby-site/**
ReactEs6Browserify/public/javascripts/basics.js
MaterialBrowserify/public/javascripts/basics.js
