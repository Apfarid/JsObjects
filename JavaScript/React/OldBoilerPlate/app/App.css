.app {
    background-color: lightblue;
    border: 1px solid blue;
}

.bap {
    background-color: red;
    height: 25px;
}
