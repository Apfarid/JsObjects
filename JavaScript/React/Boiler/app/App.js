import React from 'react';
import styles from './App.css';

export default class App extends React.Component {
    constructor(props) {
      super(props);
      this.state = {test: 'foogore One Smith'};
    }

    render() {
      return (
            <div className={styles.app}>
                bar
                <p>{this.state.test}</p>
            </div>
        );
    }
}