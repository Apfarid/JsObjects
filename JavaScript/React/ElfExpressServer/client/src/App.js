/**
 * Created by charlie on 3/31/17.
 * This version has the HTML/JSX in this component.
 */

import React from 'react';
import {createStore} from 'redux';

export default class App extends React.Component {
    constructor(props) {
        super(props);
        this.getNine = this.getNine.bind(this);
        this.setupState();
        this.state = this.store.getState();
    }

    setupState() {
        const startTitle = 'Elven Hello Express React ';
        const data = {
            origTitle: startTitle,
            title: startTitle,
            eight: '0',
            nine: '0',
            counter: 0
        };

        // Redux Reducer
        const counter = (state = data, action) => {
            switch (action.type) {
                case 'INCREMENT':
                    state.counter++;
                    break;
                case 'DECREMENT':
                    state.counter--;
                    break;
                default:
                    return state;
            }
            state.title = state.origTitle + state.counter;
            return state;
        };

        this.store = createStore(counter);
        this.store.subscribe(() => {
            const temp = this.store.getState();
            this.setState(() => ({
                counter: temp.counter,
                title: temp.title
            }));
        });

    }

    getNine() {
        /*return fetch('/getNine', {
            accept: 'application/json',
        }).then(function() {
                this.setState(() => ({
                    nine: response.json().result
                }));
            });*/
        const that = this;
        return fetch('/getNine', {
            accept: 'application/json',
        }).then(function(response) {
            return response.json();
        }).then(function(json) {
            console.log('parsed json', json);
            that.setState(() => ({nine: json.result}));
        }).catch(function(ex) {
            console.log('parsing failed', ex);
        });
        // const nine = '9999';
        // console.log('A NINE:', nine);
        // this.setState(() => ({
        //     nine: nine
        // }));
    }

    /*function checkStatus(response) {
        if (response.status >= 200 && response.status < 300) {
            return response;
        }
        const error = new Error(`HTTP Error ${response.statusText}`);
        error.status = response.statusText;
        error.response = response;
        console.log(error); // eslint-disable-line no-console
        throw error;
    }

    function parseJSON(response) {
        return response.json();
    }*/

    getEight = () => {
        this.setState(() => ({
            eight: '8'
        }));
    };

    counterUp = () => {
        this.store.dispatch({type: 'INCREMENT'});
    };

    counterDown = () => {
        this.store.dispatch({type: 'DECREMENT'});
    };

    render() {
        return (
            <div>
                <h1>{this.state.title}</h1>
                <p>
                    Welcome to {this.state.title}. This is a standard Express
                    app with React and Redux circa March, 2017.
                </p>

                <button onClick={this.getEight}>Get Eight</button>
                <button onClick={this.getNine}>Get Nine</button>


                <p>Eight: {this.state.eight}</p>
                <p>Nine: {this.state.nine}</p>

                <h2>The Countered</h2>

                <button onClick={this.counterUp}>Increment Counter</button>
                <button onClick={this.counterDown}>Decrement Counter</button>

                <p>Counter: {this.state.counter}</p>

                <h2>Debugging Tip</h2>
                <p>
                    To get started debugging your webpack, add this
                    to the <strong>webpack.config.js</strong> config file:
                </p>

                <pre>devtool: 'source-map',</pre>
                <p>Open the debugger and go webpack in the navigation pane.</p>
                <img src="debug-web.png" alt="debugging"/>
            </div>
        )
    }
}

