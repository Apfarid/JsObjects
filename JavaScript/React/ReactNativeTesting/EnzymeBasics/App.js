import React from 'react';
import {Text, View} from 'react-native';
import Address from "./Address";
import elfStyles from './elf-styles';

export default class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            foobar: 'barfoo'
        }
    }

    clickFoobar = () => {
        this.setState({foobar: 'foobar1'});
    };

    render() {
        return (
            <View style={elfStyles.container}>
                <Text id="testClosed">{this.state.foobar}</Text>
                <button id="textOpen" onClick={this.clickFoobar}>
                    Foobar
                </button>
                <Address/>
            </View>
        );
    }
}

