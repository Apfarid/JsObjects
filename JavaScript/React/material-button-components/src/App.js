import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import {blueGrey} from "@material-ui/core/colors";
import Typography from "@material-ui/core/Typography";


const useStyles = makeStyles(theme => ({
    button: {
        margin: theme.spacing(1),
    },
    input: {
        display: 'none',
    },
    paper: {
        padding: theme.spacing(3, 12),
        backgroundColor: blueGrey
    },
}));

function App() {

    const classes = useStyles();
    return (
        <div className="App">
            <Grid container spacing={3}>
                
                    <Grid item xs={12} md={12}>


                        <Grid container spacing={1} direction="column" alignItems="center">
                                <Paper className={classes.paper}>
                                <Typography variant="h1" gutterBottom>Welcome to React</Typography>
                                <Typography variant="body1" gutterBottom>With Material UI buttons and grids.</Typography>
                                </Paper>
                        </Grid>

                    </Grid>

            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12} md={12}>
                    <Grid container spacing={1} direction="column" alignItems="center">
                        <Paper className={classes.paper}>
                            <Button variant="contained" className={classes.button}>
                                Default
                            </Button>
                            <Button variant="contained" color="primary" className={classes.button}>
                                Primary
                            </Button>
                        </Paper>
                    </Grid>
                </Grid>
                <Grid item xs={12} md={12}>


                    <Grid item>
                        <Paper className={classes.paper}>
                            <ButtonGroup size="small" aria-label="Small outlined button group">
                                <Button>One Elf</Button>
                                <Button>Two Elves</Button>
                                <Button>Three Elves</Button>
                            </ButtonGroup>
                        </Paper>
                    </Grid>
                </Grid>
                <Grid item xs={12} md={12}>
                    <Paper className={classes.paper}>
                        <Grid container spacing={1} direction="column" alignItems="center">
                            <Grid item>
                                <ButtonGroup variant="contained" size="small" aria-label="Small contained button group">
                                    <Button>Elf One</Button>
                                    <Button>Elf Two</Button>
                                    <Button>Elf Three</Button>
                                </ButtonGroup>
                            </Grid>
                            <Grid item>
                                <ButtonGroup
                                    variant="contained"
                                    color="primary"
                                    aria-label="Full-width contained primary button group"
                                >
                                    <Button>Elven One</Button>
                                    <Button>Elven Two</Button>
                                    <Button>Elven Three</Button>
                                </ButtonGroup>
                            </Grid>
                            <Grid item>
                                <ButtonGroup
                                    variant="contained"
                                    color="secondary"
                                    size="large"
                                    aria-label="Large contained secondary button group"
                                >
                                    <Button>Alpha</Button>
                                    <Button>Beta</Button>
                                    <Button>Echo</Button>
                                </ButtonGroup>
                            </Grid>
                        </Grid>
                    </Paper>
                </Grid>
            </Grid>
        </div>
    );

}

export default App;
