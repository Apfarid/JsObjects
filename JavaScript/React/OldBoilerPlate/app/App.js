import React from 'react';
import './App.css';

export default class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = { test: 'Test state' };
    }

    render() {
        return (
            <div className="app">
                <h1>Old Boilerplate for React</h1>
                <p>
                    This was one of my first tries at using at react. It was
                    based on an{' '}
                    <a
                        href="https://github.com/christianalfoni/webpack-express-boilerplate"
                        target="_blank"
                    >
                        abandoned git project
                    </a>
                    . I've updated it (March 2018) and kept it around because it
                    has a few useful bits in it. However, I would recommend
                    basing new applications on{' '}
                    <a
                        href="https://github.com/facebook/create-react-app"
                        target="_blank"
                    >
                        create-react-app
                    </a>
                    . Though <strong>create-react-app</strong> is usually the
                    best option, if you want to understand the basics, try my{' '}
                    <a
                        href="https://github.com/charliecalvert/JsObjects/tree/master/JavaScript/React/ReactBasics"
                        target="_blank"
                    >
                        ReactBasics
                    </a>
                    . It is much easier to understand than either{' '}
                    <strong>create-react-app</strong> or this app.
                </p>
                <p>{this.state.test}</p>
            </div>
        );
    }
}
