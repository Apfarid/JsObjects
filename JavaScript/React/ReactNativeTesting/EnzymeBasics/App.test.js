import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('App tests', function() {

    const debug = false;

    it('renders without crashing', () => {
        const rendered = renderer.create(<App/>).toJSON();
        expect(rendered).toBeTruthy();
    });

    it('renders default output with react (mount)', () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click on foobar with react', () => {
        const after = renderer.create(<App/>);
        after.getInstance().clickFoobar();
        expect(after.toJSON()).toMatchSnapshot();
    });

    it('renders default output with enzyme shallow', () => {
        const wrapper = shallow(<App/>);
        if (debug) {
            console.log(wrapper.debug());
        }
        expect(wrapper).toMatchSnapshot();

    });

    it('handles click on foobar with enzyme', () => {
        const wrapper = shallow(<App/>);
        if (debug) {
            console.log(wrapper.debug());
        }
        wrapper.find('#textOpen').simulate('click');
        expect(wrapper).toMatchSnapshot();
    });
});