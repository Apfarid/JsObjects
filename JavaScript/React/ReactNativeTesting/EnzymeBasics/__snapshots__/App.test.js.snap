// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests handles click on foobar with enzyme 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    id="testClosed"
  >
    foobar1
  </Text>
  <button
    id="textOpen"
    onClick={[Function]}
  >
    Foobar
  </button>
  <Address />
</Component>
`;

exports[`App tests handles click on foobar with react 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    id="testClosed"
  >
    foobar1
  </Text>
  <button
    id="textOpen"
    onClick={[Function]}
  >
    Foobar
  </button>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Address
    </Text>
  </View>
</View>
`;

exports[`App tests renders default output with enzyme shallow 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    id="testClosed"
  >
    barfoo
  </Text>
  <button
    id="textOpen"
    onClick={[Function]}
  >
    Foobar
  </button>
  <Address />
</Component>
`;

exports[`App tests renders default output with react (mount) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    id="testClosed"
  >
    barfoo
  </Text>
  <button
    id="textOpen"
    onClick={[Function]}
  >
    Foobar
  </button>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Address
    </Text>
  </View>
</View>
`;
